src/ILCompiler.ReadyToRun/src/JitInterface/CorInfoImpl.ReadyToRun.cs

```diff
@@ -151,6 +151,7 @@ private static mdToken FindGenericMethodArgTypeSpec(EcmaModule module)
         public void CompileMethod(IReadyToRunMethodCodeNode methodCodeNodeNeedingCode)
         {
             _methodCodeNode = methodCodeNodeNeedingCode;
+            _profileDataNode = null;
```

Comment:
We clean up state in `CompileMethodCleanup()`

Reply:
Moved